src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.TextWriter.cs

```diff
@@ -0,0 +1,489 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="ODataUtf8JsonWriter.TextWriter.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+#if NETCOREAPP
+namespace Microsoft.OData.Json
+{
+    using System;
+    using System.Buffers;
+    using System.Diagnostics;
+    using System.IO;
+    using System.Text;
+    using System.Text.Unicode;
+    using System.Threading.Tasks;
+
+    internal sealed partial class ODataUtf8JsonWriter
+    {
+        /// <summary>
+        /// Starts a TextWriter value scope with the specified content type.
+        /// </summary>
+        /// <param name="contentType">The content type of the TextWriter value scope.</param>
+        /// <returns>A TextWriter for writing JSON data.</returns>
+        public TextWriter StartTextWriterValueScope(string contentType)
+        {
+            this.CommitUtf8JsonWriterContentsToBuffer();
+
+            WriteItemWithSeparatorIfNeeded();
+
+            this.currentTextWriterContentType = contentType;
+            bool isWritingJson = String.IsNullOrEmpty(this.currentTextWriterContentType) || this.currentTextWriterContentType.StartsWith(MimeConstants.MimeApplicationJson, StringComparison.Ordinal);
+
+            // We don't perform escaping when the content type is not JSON
+            // since we assume the input to be already a valid JSON with all escaping handled by the caller.
+            if (!isWritingJson)
+            {
+                this.bufferWriter.Write(this.DoubleQuote.Slice(0, 1).Span);
+            }
+
+            this.Flush();
+
+            this.textWriter = new ODataUtf8JsonTextWriter(this, isWritingJson);
+
+            return this.textWriter;
+        }
+
+        /// <summary>
+        /// Ends a TextWriter value scope.
+        /// </summary>
+        public void EndTextWriterValueScope()
+        {
+            bool isWritingJson = String.IsNullOrEmpty(this.currentTextWriterContentType) || this.currentTextWriterContentType.StartsWith(MimeConstants.MimeApplicationJson, StringComparison.Ordinal);

```

Comment:
nit:
```suggestion
            bool isWritingJson = string.IsNullOrEmpty(this.currentTextWriterContentType) || this.currentTextWriterContentType.StartsWith(MimeConstants.MimeApplicationJson, StringComparison.Ordinal);
```
It's very likely the analyzer is flagging this...